tests/jobs/localisation/test_integration.py

```diff
@@ -0,0 +1,252 @@
+import yaml
+import pytest
+from res.enkf import EnKFMain, ResConfig
+from semeio.workflows.localisation.local_config_script import LocalisationConfigJob
+from xtgeo.surface.regular_surface import RegularSurface
+import xtgeo
+import numpy as np
+
+
+@pytest.mark.parametrize(
+    "obs_group_add, param_group_add, expected, expected_obs1, expected_obs2",
+    [
+        (
+            ["FOPR", "WOPR_OP1_190"],
+            ["SNAKE_OIL_PARAM:OP1_DIVERGENCE_SCALE", "SNAKE_OIL_PARAM:OP1_OFFSET"],
+            ["SNAKE_OIL_PARAM"],
+            [
+                "WOPR_OP1_108",
+                "WOPR_OP1_144",
+                "WOPR_OP1_36",
+                "WOPR_OP1_72",
+                "WOPR_OP1_9",
+                "WPR_DIFF_1",
+            ],
+            ["FOPR", "WOPR_OP1_190"],
+        ),
+    ],
+)
+def test_localisation(
+    setup_ert, obs_group_add, param_group_add, expected, expected_obs1, expected_obs2
+):
+    ert = EnKFMain(setup_ert)
+    config = {
+        "correlations": [
+            {
+                "name": "CORR1",
+                "obs_group": {"add": "*", "remove": obs_group_add},
+                "param_group": {
+                    "add": "SNAKE_OIL_PARAM:OP1_PERSISTENCE",
+                },
+            },
+            {
+                "name": "CORR2",
+                "obs_group": {"add": "*", "remove": obs_group_add},
+                "param_group": {
+                    "add": "SNAKE_OIL_PARAM:*",
+                    "remove": "SNAKE_OIL_PARAM:OP1_PERSISTENCE",
+                },
+            },
+            {
+                "name": "CORR3",
+                "obs_group": {"add": obs_group_add},
+                "param_group": {
+                    "add": param_group_add,
+                },
+            },
+        ],
+    }
+    with open("local_config.yaml", "w") as fout:
+        yaml.dump(config, fout)
+    LocalisationConfigJob(ert).run("local_config.yaml")
+    assert ert.getLocalConfig().getMinistep("CORR1").name() == "CORR1"
+    assert (
+        ert.getLocalConfig().getObsdata("CORR1_obs_group").name() == "CORR1_obs_group"
+    )
+    assert len(ert.getLocalConfig().getUpdatestep()) == 3
+    ministep_names = ["CORR1", "CORR2", "CORR3"]
+    for index, ministep in enumerate(ert.getLocalConfig().getUpdatestep()):
+        assert ministep.name() == ministep_names[index]
+        obs_list = []
+        for count, obsnode in enumerate(ministep.getLocalObsData()):
+            obs_list.append(obsnode.key())
+        obs_list.sort()
+
+        if index in [0, 1]:
+            assert obs_list == expected_obs1
+        else:
+            assert obs_list == expected_obs2
+        key = ministep_names[index] + "_param_group"
+        assert ministep[key].keys() == expected
```

Comment:
Because you are not using parameterize for running more than one case here, I would suggest you rewrite this slightly, my suggestion would be something like this:

```suggestion
@pytest.mark.parametrize(
    "obs_group_add, param_group_add, expected",
    [
        (
            ["FOPR", "WOPR_OP1_190"],
            ["SNAKE_OIL_PARAM:OP1_DIVERGENCE_SCALE", "SNAKE_OIL_PARAM:OP1_OFFSET"],
            ["SNAKE_OIL_PARAM"],
        ),
    ],
)
def test_localisation(
    setup_ert, obs_group_add, param_group_add, expected
):
    ert = EnKFMain(setup_ert)
    config = {
        "correlations": [
            {
                "name": "CORR1",
                "obs_group": {"add": "*", "remove": obs_group_add},
                "param_group": {
                    "add": "SNAKE_OIL_PARAM:OP1_PERSISTENCE",
                },
            },
            {
                "name": "CORR2",
                "obs_group": {"add": "*", "remove": obs_group_add},
                "param_group": {
                    "add": "SNAKE_OIL_PARAM:*",
                    "remove": "SNAKE_OIL_PARAM:OP1_PERSISTENCE",
                },
            },
            {
                "name": "CORR3",
                "obs_group": {"add": obs_group_add},
                "param_group": {
                    "add": param_group_add,
                },
            },
        ],
    }
    with open("local_config.yaml", "w") as fout:
        yaml.dump(config, fout)
    LocalisationConfigJob(ert).run("local_config.yaml")
    assert ert.getLocalConfig().getMinistep("CORR1").name() == "CORR1"
    assert (
        ert.getLocalConfig().getObsdata("CORR1_obs_group").name() == "CORR1_obs_group"
    )
    result = {}
    for index, ministep in enumerate(ert.getLocalConfig().getUpdatestep()):
        result[ministep.name()] = {
            "obs": [obs_node.key() for obs_node in ministep.getLocalObsData()],
            "key": ministep.name() + "_param_group",
        }

    expected_result = {
        "CORR1": {
            "obs": [
                "WOPR_OP1_108",
                "WOPR_OP1_144",
                "WOPR_OP1_36",
                "WOPR_OP1_72",
                "WOPR_OP1_9",
                "WPR_DIFF_1",
            ],
            "key": "CORR1_param_group",
        },
        "CORR2": {
            "obs": [
                "WOPR_OP1_108",
                "WOPR_OP1_144",
                "WOPR_OP1_36",
                "WOPR_OP1_72",
                "WOPR_OP1_9",
                "WPR_DIFF_1",
            ],
            "key": "CORR2_param_group",
        },
        "CORR3": {"obs": ["FOPR", "WOPR_OP1_190"], "key": "CORR3_param_group"},
    }
    assert result == expected_result
```

It is always good to avoid `assert` in loops where possible, as that can obscure what the error really is, and `pytest` is really good at showing comparisons when asserting on lists and dicts. This can easily be converted back in to a parameterized test if you would like to add another test.

Reply:
I have included your version of this test, but I am not sure I should remove the original test since it actually loops through the created ministeps and get info from those and verify that the created ministeps contain what they should. I thought thtat could be a good test to check that the ministeps are initialized as they should. By keeping both your version and the original the simplicity is take care of by your test and the check for the state of the ministep objects by the original test